src/cli/tests/cli.test.ts

```diff
@@ -0,0 +1,370 @@
+      } else {
+        resolve({ stdout, stderr });
+      }
+    });
+  });
+}
+
+async function mintEthToAccount(address: string): Promise<{ stdout: string; stderr: string }> {
+  const res = await sh(
+    `curl localhost:5050/mint -H "Content-Type: application/json" -d "{ \\"address\\": \\"${address}\\", \\"amount\\": 1000000000000000000, \\"lite\\": false }"`,
+  );
+  return res;
+}
+
+describe('Manage starknet account', function () {
+  this.timeout(TIME_LIMIT);
+  it('should create a starknet account', async () => {
+    if (fs.existsSync(path.resolve(__dirname, 'starknet_open_zeppelin_accounts.json'))) {
+      fs.unlinkSync(path.resolve(__dirname, 'starknet_open_zeppelin_accounts.json'));
+    }
+
+    const { stdout, stderr } = await sh(
+      `${warpBin} new_account --wallet ${starknet_wallet} --network ${network} --feeder_gateway_url ${gateway_url} --account_dir ${account_dir}`,
+    );
+
+    expect(stderr).to.be.empty;
+    expect(stdout).to.not.be.empty;
+
+    const starknet_open_zeppelin_accounts = JSON.parse(
+      fs.readFileSync(path.resolve(__dirname, 'starknet_open_zeppelin_accounts.json'), 'utf-8'),
+    );
+
+    // verify the properties of the created account
+    expect(starknet_open_zeppelin_accounts[network]).to.not.be.undefined;
+    expect(starknet_open_zeppelin_accounts[network]['__default__']).to.not.be.undefined;
+    expect(starknet_open_zeppelin_accounts[network]['__default__']['private_key']).to.not.be
+      .undefined;
+    expect(starknet_open_zeppelin_accounts[network]['__default__']['public_key']).to.not.be
+      .undefined;
+    expect(starknet_open_zeppelin_accounts[network]['__default__']['salt']).to.not.be.undefined;
+    expect(starknet_open_zeppelin_accounts[network]['__default__']['address']).to.not.be.undefined;
+    expect(starknet_open_zeppelin_accounts[network]['__default__']['deployed']).to.be.false;
+  });
+
+  it('mint wei to the generated account', async () => {
+    const starknet_open_zeppelin_accounts = JSON.parse(
+      fs.readFileSync(path.resolve(__dirname, 'starknet_open_zeppelin_accounts.json'), 'utf-8'),
+    );
+    const address = starknet_open_zeppelin_accounts[network]['__default__']['address'];
+
+    const { stdout } = await mintEthToAccount(address);
+
+    const response = JSON.parse(stdout);
+
+    expect(response).to.not.be.undefined;
+    expect(response['new_balance']).to.not.be.undefined;
+    expect(response['new_balance']).to.equal(1000000000000000000);
+    expect(response['tx_hash']).to.not.be.undefined;
+  });
+
+  it('should deploy the starknet account', async () => {
+    const { stdout, stderr } = await sh(
+      `${warpBin} deploy_account --wallet ${starknet_wallet} --feeder_gateway_url ${gateway_url} --gateway_url ${gateway_url} --network ${network} --account_dir ${account_dir}`,
+    );
+
+    expect(stderr).to.be.empty;
+    expect(stdout).to.not.be.empty;
+
+    const contract_address = stdout.split('\n')[3].split(':')[1].trim();
```

Comment:
I think if make a function that takes the txHash and contractAddress and returns the full string and we can do a exact string match that will be a lot more readable than string splitting on new lines and colons.

Reply:
This applies to other string splitting matchers as well

Reply:
we are extracting txHash and contractAddress from the output itself

Reply:
Creating a function may not be the most efficient solution, as the output may vary based on the fee applied and the context of the message.

Reply:
Creating a function may not be the most efficient solution, as the output may vary based on the fee applied and the context of the message. Each time we would need to analyze the entire string and then compare the analyzed string to the expected format.

As sample output string looks as : 
```
Running starknet compile with cairoPath /Users/rohit/nmd/warp
Sending the transaction with max_fee: 0.001247 ETH (1246520000000000 WEI).
Invoke transaction for contract deployment was sent.
Contract address: 0x05a252845cba4af66f37d1c759027049a8592b8a7ad29db567d16e96fee24ea7
Transaction hash: 0x15e02734241afc4cd37578ff160c313fb41562446ad40ea92992b4c6e36fe84

```

Reply:
Is the `max_fee` not static in all the test cases?

Reply:
In that case, instead of string splitting, please convert it to a regex and use matchAll

Reply:
I'm not using max_fee, but anyway I'm adding some changes to improve readability and output string validation